File: solutions/JavaScript/136.SingleNumber.js
var singleNumber = function (nums) {
    if (nums.length <= 1) return nums[0];

    let literal = {};

    for (num of nums) {
        if (literal[num] === undefined) {
            literal[num] = 1;
        } else {
            if (literal[num]) literal[num] += 1;
        }
    }

    for (let key in literal) {
        if (literal[key] === 1) {
            return Number(key);
        }
    }
};

File: solutions/Python/136.SingleNumber.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        if len(nums) <= 1:
            return nums[0]

        dictionary = {}
        for num in nums:
            if dictionary.get(num) == None:
                dictionary[num] = 1
            else:
                dictionary[num] += 1

        for key, value in dictionary.items():
            if value == 1:
                return key
